tests/check_pip

```diff
@@ -161,62 +206,125 @@
-		4) printf "%s\n" "FAIL: pip-audit detected security vulnerabilities." >&2 ;;
-		5) printf "%s\n" "FAIL: pip-licenses detected license issues." >&2 ;;
-		6) printf "%s\n" "FAIL: pip install failed." >&2 ;;
-		126) printf "%s\n" "SKIP: Unable to continue script execution." >&2 ;;
-		*) printf "%s\n" "FAIL: Detected requirements errors." >&2 ;;
+		0) printf "::notice title=OK::%s\n" "OK: Found no detected requirements errors." ;;
+		1) printf "::error file=${SCRIPT_FILE},line=${BASH_LINENO:-0},title=CHECK-PIP::%s\n" "FAIL: General failure during script execution." >&2 ;;
+		3) printf "::error file=${SCRIPT_FILE},line=${BASH_LINENO:-0},title=CONFIGURATION::%s\n" "FAIL: Gathering repostory's requirements failed." >&2 ;;  # git ls-tree command failed
+		4) printf "::error file=${SCRIPT_FILE},line=${BASH_LINENO:-0},title=SECURITY::%s\n" "FAIL: pip-audit detected security vulnerabilities." >&2 ;;
+		5) printf "::error file=${SCRIPT_FILE},line=${BASH_LINENO:-0},title=LICENSE::%s\n" "FAIL: pip-licenses detected license issues." >&2 ;;
+		6) printf "::error file=${SCRIPT_FILE},line=${BASH_LINENO:-0},title=INSTALL::%s\n" "FAIL: pip install failed." >&2 ;;
+		126) printf "::warning file=${SCRIPT_FILE},line=${BASH_LINENO:-0},title=SKIPPED::%s\n" "SKIP: Unable to continue script execution." >&2 ;;
+		*) printf "::error file=${SCRIPT_FILE},line=${BASH_LINENO:-0},title=FAILED::%s\n" "FAIL: Detected requirements errors." >&2 ;;
 	esac
+	printf "::endgroup::\n" ;
 }
 
 function navigate_dirs_by_git() {
 	if _TEST_ROOT_DIR=$(git rev-parse --show-superproject-working-tree 2>/dev/null); then
 		if [ -z "${_TEST_ROOT_DIR}" ]; then
 			_TEST_ROOT_DIR=$(git rev-parse --show-toplevel 2>/dev/null)
 		fi
+		printf "::debug::%s\n" "Found ${_TEST_ROOT_DIR} ..." ;
 	else
-		printf "\t%s\n" "FAIL: missing valid repository or source structure" >&2
+		printf "::error file=${SCRIPT_FILE},line=${BASH_LINENO:-0},title=${FUNCNAME:-$0}::%s\n" "FAIL: missing valid repository or source structure" >&2
 		EXIT_CODE=40
 	fi
 }
 
+function check_license_when_given_req() {
+	local req_file_input="${1}"
+	shift ;  # allows $@ to work as before
+	local SUB_CODE=${EXIT_CODE-0}
+	umask 007
+	printf "::debug::%s\n" "need venv ..." ;
+	# Create a temporary directory for the virtual environment
+	temp_dir=$(mktemp -d)
+
+	# Enter the temporary directory
+	cd "$temp_dir"
+
+	# Create a virtual environment using venv
+	python3 -m venv venv
+
+	# Activate the virtual environment
+	source venv/bin/activate
+
+	umask 037
+	# 2>&1 >/dev/null
+	python3 -m pip install $PIP_COMMON_FLAGS $PIP_ENV_FLAGS "pip-licenses>=5.0" || SUB_CODE=6 ;
+	wait ;
+	printf "::debug::%s\n" "venv setup ... (${SUB_CODE})" ;
+	# Install the given Python modules using pip
+	for module in $@ ; do
+		printf "::debug::%s\n" "Checking license from package '$module' ..." ;
+		REQ_SPEC=$(grep -F "$module" <(cat <"${_TEST_ROOT_DIR}"/$req_file | sed -E -e '/^[[:space:]]*$/d' | sed -E -e '/^[#]+.*$/d') | grep -m1 -F "$module" )
```

Comment:
😕 sounds interesting; please elaborate what a command injection is in this case for posterity, and then explain how it could be accomplished with the current implementation.

Reply:
open a followup issue for this